src/detect-content.c

```diff
@@ -112,9 +113,18 @@ int DetectContentDataParse(const char *keyword, const char *contentstr,
                 bin_count++;
                 if (bin) {
                     if (binpos > 0) {
-                        SCLogError(SC_ERR_INVALID_SIGNATURE,
-                                "Incomplete hex code in content - %s. Invalidating signature.",
+                        SCLogWarning(SC_ERR_INVALID_SIGNATURE,
```

Comment:
I think this should only be a warning if we're not in strict mode. If we are in strict mode it should be an error like it was.